inst/tests/nafill.Rraw

```diff
@@ -203,3 +204,127 @@ test(7.07, setnafill(DT, fill=0, cols=1L), copy(DT)[ , a := ans1])
+}
+
+# coerceAs verbose
+options(datatable.verbose=2L)
+input = 1
+test(10.01, ans<-coerceAs(input, 1), 1, output="double[numeric] into double[numeric]")
+test(10.02, address(input)!=address(ans))
+test(10.03, ans<-coerceAs(input, 1, copy=FALSE), 1, output="copy=false and input already of expected type and class double[numeric]")
+test(10.04, address(input), address(ans))
+test(10.05, ans<-coerceAs(input, 1L), 1L, output="double[numeric] into integer[integer]")
+test(10.06, address(input)!=address(ans))
+test(10.07, ans<-coerceAs(input, 1L, copy=FALSE), 1L, output="double[numeric] into integer[integer]", notOutput="copy=false")
+test(10.08, address(input)!=address(ans))
+test(10.09, coerceAs("1", 1L), 1L, output="character[character] into integer[integer]")
+test(10.10, coerceAs("1", 1), 1, output="character[character] into double[numeric]")
+test(10.11, coerceAs("a", factor("x")), factor("a", levels=c("x","a")), output="character[character] into integer[factor]") ## levels of 'as' are retained!
+test(10.12, coerceAs("a", factor()), factor("a"), output="character[character] into integer[factor]")
+test(10.13, coerceAs(1, factor("x")), factor("x"), output="double[numeric] into integer[factor]")
+test(10.14, coerceAs(1, factor("x", levels=c("x","y"))), factor("x", levels=c("x","y")), output="double[numeric] into integer[factor]")
+test(10.15, coerceAs(2, factor("x", levels=c("x","y"))), factor("y", levels=c("x","y")), output="double[numeric] into integer[factor]")
+test(10.16, coerceAs(1:2, factor(c("x","y"))), factor(c("x","y")), output="integer[integer] into integer[factor]")
+test(10.17, coerceAs(1:3, factor(c("x","y"))), output="integer[integer] into integer[factor]", error="factor numbers.*3 is outside the level range")
+test(10.18, coerceAs(c(1,2,3), factor(c("x","y"))), output="double[numeric] into integer[factor]", error="factor numbers.*3.000000 is outside the level range")
+test(10.19, coerceAs(factor("x"), factor(c("x","y"))), factor("x", levels=c("x","y")), output="integer[factor] into integer[factor]")
+test(10.20, coerceAs(factor("x"), factor(c("x","y")), copy=FALSE), factor("x", levels=c("x","y")), output="input already of expected type and class") ## copy=F has copyMostAttrib
+a = structure("a", class="a")
+b = structure("b", class="b")
+test(10.21, coerceAs(a, b), structure("a", class="b"), output="character[a] into character[b]")
+a = structure(1L, class="a")
+b = structure(2L, class="b")
+test(10.22, coerceAs(a, b), structure(1L, class="b"), output="integer[a] into integer[b]")
+a = structure(1, class="a")
+b = structure(2, class="b")
+test(10.23, coerceAs(a, b), structure(1, class="b"), output="double[a] into double[b]")
+a = structure(1, class="a")
+b = structure(2L, class="b")
+test(10.24, coerceAs(a, b), structure(1L, class="b"), output="double[a] into integer[b]")
+if (test_bit64) {
+  x = as.integer64(1L)
+  test(10.81, coerceAs(x, 1), 1, output="double[integer64] into double[numeric]")
```

Comment:
can we add some tests with `NA` (coercing to/from `NA` in the int64 types)? I'm not sure `memrecycle` does the trick for these because there's no conversion `NA_INTEGER64 <-> NA_REAL` right?

Reply:
AFAIR those are above near `# coerceAs int/numeric/int64 as used in nafill`